File: solutions/Java/palindrome_number_without_string.java
public class Palindrome {
    public static boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        if (x == 0) {
            return true;
        }
        if (x % 10 == 0) {
            return false;
        }
        int numReversed = 0;
        int originalX = x;
        while (originalX > 0) {
            int lastDigit = originalX % 10;
            numReversed = numReversed * 10 + lastDigit;
            originalX /= 10;
        }
        return x == numReversed;
    }

    public static void main(String[] args) {
        System.out.println(isPalindrome(121));  // true
        System.out.println(isPalindrome(-121)); // false
        System.out.println(isPalindrome(10));   // false
    }
}

File: solutions/Python/palindrome_number_without_string.py
def is_palindrome(x: int) -> bool:
    if x < 0:
        return False
    if x == 0:
        return True
    if x % 10 == 0:
        return False
    num_reversed = 0
    original_x = x
    while original_x > 0:
        last_digit = original_x % 10
        num_reversed = num_reversed * 10 + last_digit
        original_x //= 10
    return x == num_reversed

print(is_palindrome(121))  # True
print(is_palindrome(-121))  # False
print(is_palindrome(10))  # False
